server/implementation/src/main/java/io/smallrye/graphql/execution/ExecutionDecorator.java

```diff
@@ -0,0 +1,18 @@
+package io.smallrye.graphql.execution;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+
+public interface ExecutionDecorator {
+    default void before(ExecutionInput executionInput) {
```

Comment:
Out of curiosity, why are these methods all `default`?

Also, not necessary, but maybe we could make `before` and `after` non-default and then make `onError` default and implement it by just delegating to `after`?  For Tracing the decorator definitely wants to distinguish between a normal result and an error result, but maybe in the future other decorators won't care - and just want to log entry/exit (regardless of the success of the execution).  Just a thought - it's completely fine the way it is.

Reply:
Had the same thought that some decorators don't need everything, but to have all of them as default could indeed be too much.

Maybe it is also useful to add an additional `finally` method to have a single exit point?Normally it will call `before`, `after` (maybe renamed) and `finally`, and in case of an error `before`, `onError` and `finally`. `after` and `onError` would then be default. 

The same might also be done for `DataFetcherDecorator`.

Reply:
It's probably fine the way it is now. One nice thing about all methods being default is that if somebody were to add a new ExecutionDecorator that only needed to do something before the invocation or only in an error scenario, etc., then they only need to write one method.

If we did need to add a new decorator that always executed regardless of error or successful execution, we could add that in later - either a `finally` method or change the default implementation of `onError` to delegate to `after`.  But until we need to handle that scenario, I don't think we need to change anything with this PR.  

Nice job!